src/gui/rss/rsswidget.cpp

```diff
@@ -375,6 +382,89 @@ void RSSWidget::downloadSelectedTorrents()
     }
 }
 
+// Provides a way to filter a torrent's name. Used for Right-Click -> Add new RSS Rule.
+const QString RSSWidget::sanitizeNewRssRuleName(const QString &ruleName)
+{
+    QString name;
+
+    for (QChar currentChar : ruleName)
+    {
+        if (currentChar.isSpace() || currentChar.isLetterOrNumber())
+        {
+            name.append(currentChar);
+        }
+    }
+
+    return name.simplified(); // return without multiple whitespace
+}
+
+void RSSWidget::newRSSRuleBtn() {
+
+    // Safety:
+    // The feature only appears if there is only 1 article selected,
+    // but just to make sure, we do another check here.
+    if (m_articleListWidget->selectedItems().count() != 1)
+        return;
+
+    // Retrieve article from selected items
+    QListWidgetItem *selectedRssRule = asConst(m_articleListWidget->selectedItems()[0]);
+    RSS::Article *selectedRssRuleArticle = selectedRssRule->data(Qt::UserRole).value<RSS::Article *>();
+    Q_ASSERT(selectedRssRuleArticle);
+
+    // Get the title and also sanitize it
+    QString selectedRssRuleArticleTitle = selectedRssRuleArticle->title();
+    QString sanitizeRSSRuleName = sanitizeNewRssRuleName(selectedRssRuleArticleTitle);
```

Comment:
Is sanitization really required? What kind of unacceptable characters do you intend to filter out in common case?

Reply:
Well, it might be Japanese characters, lot of [] and whatnot, it just results in a super messy rule name. Maybe qBittorrent, the code, and Qt can deal with all that. Not a huge problem, just the copy-paste name looked a bit silly.

But, the suggestion (just below) to just name new rules "New Rule" would eliminate all this silliness.

One question regarding this though: Is it OK, if it's just "New Rule" in every language? I don't mind, just asking.

Reply:
> But, the suggestion (just below) to just name new rules "New Rule" would eliminate all this silliness.

👍 


> One question regarding this though: Is it OK, if it's just "New Rule" in every language? I don't mind, just asking.

Why not make it translatable?

But now I have general comment about feature design. I dislike the unnecessary dependencies between components that your implementation creates. I believe it could be better to do it in another way by delegate all the jobs to AutomatedRssDownloader. You could add yet another constructor that accepts `RSS::Article` as parameter that additionally to regular job will add new Rule (using its regular way) and fill it with initial data based on given article).

Reply:
Hmm I tried to do this today, but the constructor seems rather "complex" in AutomatedRssDownloader. You sure about the new constructor idea?

Like, then there are two constructors, so if someone has to touch one of them, the other one also has to be modified, etc. Basically the logic would be duplicated. Or, put all logic (that I just can) into a new function and then both constructors call that. Maybe. 

Sorry for making this simple PR such a tedious chore btw, feel free to throw it if it looks more hassle than what it's worth.

Reply:
>You sure about the new constructor idea?

Yes.
>put all logic (that I just can) into a new function and then both constructors call that.

In fact, everything is even simpler. In modern C++, you can call a constructor in the initialization list of another constructor, so all you need to do is do the following:
```c++
// automatedrssdownloader.h
explicit AutomatedRssDownloader(RSS::Article *newRuleTemplate, QWidget *parent = nullptr);

// automatedrssdownloader.cpp
AutomatedRssDownloader::AutomatedRssDownloader(RSS::Article *newRuleTemplate, QWidget *parent)
    : AutomatedRssDownloader(parent)
{
    // Additional logic here...
}
```